Lib/test/test_typing.py

```diff
@@ -7679,6 +7679,71 @@ def test_bad_var_substitution(self):
                 with self.assertRaises(TypeError):
                     collections.abc.Callable[P, T][arg, str]
 
+    def test_type_var_subst_for_other_type_vars(self):
```

Comment:
These tests look really great. Could we also add one or two cases for classes that are generic over a `ParamSpec` and a `TypeVar`? E.g.:

```python
class Base2(Generic[P, T]):
    pass

A11 = Base2[[int, T2], T3]
A12 = Base2[[int, *Ts], T2]
```

Reply:
Done, I've added even more corner cases.